packages/commcare/src/Utils.js

```diff
@@ -0,0 +1,51 @@
+import { composeNextState } from '@openfn/language-common';
+import {
+  request as commonRequest,
+  makeBasicAuthHeader,
+  logResponse,
+} from '@openfn/language-common/util';
+
+export const prepareNextState = (state, response, callback = s => s) => {
+  const { body, ...responseWithoutBody } = response;
+  const nextState = {
+    ...composeNextState(state, response.body),
+    response: responseWithoutBody,
+  };
+
+  return callback(nextState);
+};
+
+export function request({
+  state,
+  method,
+  path,
+  data,
+  params = {},
+  header = {},
+  authType,
+  contentType = 'application/json',
+  parseAs = 'json',
+}) {
+  const { hostUrl, username, password } = state.configuration;
```

Comment:
Ah this is why we need to pass state. hmm.

Ok, maybe we'll address this in the auth issue #554. Ignore my comments on state for now

Reply:
Basically I think we should pass an `auth` object, which is basically state.config, and we decide what auth to used based on the keys.